### PR TITLE
msg: ignore request_redirect_t encode/decode when we not need

### PR DESCRIPTION
**Pull request by XinzeChi**
reusing the feature introduced by commit 39e06ef

Signed-off-by: Xinze Chi <xinze@xsky.com>

### CODE REVIEW

**Comment by XinzeChi** [comment]
@liewegas , is it make sense ?

File: src/messages/MOSDOpReply.h [hunk 1]
```diff
@@ -189,7 +191,15 @@ class MOSDOpReply : public Message {
 
       ::encode(replay_version, payload);
       ::encode(user_version, payload);
-      ::encode(redirect, payload);
+      if ((features & CEPH_FEATURE_NEW_OSDOPREPLY_ENCODING) == 0) {
+        ::encode(redirect, payload);
```
Comment (liewegas) [inline src/messages/MOSDOpReply.h]: you need to set head.version = 6 here

**Comment by liewegas** [comment]
Please rebase on master, too.  thanks!

**Comment by XinzeChi** [comment]
@liewegas , Please review again. Thanks.

**Comment by liewegas** [comment]
lgtm!

**Comment by liewegas** [comment]
please rebase?

**Comment by XinzeChi** [comment]
rebase @liewegas

**Comment by liewegas** [comment]
replaced by #7386

**Pull request closed without merging**